### ISSUE TITLE
False Positive on Lint/LiteralAssignmentInCondition

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#12434]
## Expected behavior

Lint/LiteralAssignmentInCondition should not register for non-literals

## Actual behavior

A string with interpolation in the condition of a loop where the interpolation values change inside is triggering the cop.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

def my_method
  until block.call(new_name = "#{dir}#{basename}-#{addition}#{extname}")
    addition += 1
    return random_backup_name if addition >= 8
  end
  new_name
end
```

`rubocop --only  Lint/LiteralAssignmentInCondition test.rb`

## RuboCop version

```
% rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.1.3) [arm64-darwin21]
```
